test/test.path.ts

```diff
@@ -113,40 +115,100 @@ describe('Path params', () => {
-    assert.equal(req.uri.pathname, '/drive/v2/files/123abc');
+  it('should be put in URL of pathname', (done) => {
+    const p = '/drive/v2/files/123abc';
+    nock(Utils.baseUrl).get(p).reply(200);
+    localDrive.files.get({fileId: '123abc'}, (err, res) => {
+      if (err) {
+        return done(err);
+      }
+      assert.equal(url.parse(res.config.url).path, p);
+      nock(Utils.baseUrl).get(p).reply(200);
+      remoteDrive.files.get({fileId: '123abc'}, (err2, res2) => {
+        if (err2) {
+          return done(err2);
+        }
+        assert.equal(url.parse(res.config.url).path, p);
+        done();
+      });
+    });
   });
 
-  it('should not be urlencoded', () => {
-    let req = localDrive.files.get({fileId: 'p@ram'}, Utils.noop);
-    assert.equal(req.uri.path.split('/').pop(), 'p@ram');
-    req = remoteDrive.files.get({fileId: 'p@ram'}, Utils.noop);
-    assert.equal(req.uri.path.split('/').pop(), 'p@ram');
+  it('should not be urlencoded', (done) => {
+    const p = '/drive/v2/files/p@ram';
+    nock(Utils.baseUrl).get(p).reply(200);
+    localDrive.files.get({fileId: 'p@ram'}, (err, res) => {
+      if (err) {
+        return done(err);
+      }
+      assert.equal(url.parse(res.config.url).path.split('/').pop(), 'p@ram');
+      nock(Utils.baseUrl).get(p).reply(200);
+      remoteDrive.files.get({fileId: 'p@ram'}, (err2, res2) => {
+        if (err2) {
+          return done(err2);
+        }
+        assert.equal(url.parse(res.config.url).path.split('/').pop(), 'p@ram');
+        done();
+      });
+    });
   });
 
-  it('should keep query params null if only path params', () => {
-    let req = localDrive.files.get({fileId: '123abc'}, Utils.noop);
-    assert.equal(req.uri.query, null);
-    req = remoteDrive.files.get({fileId: '123abc'}, Utils.noop);
-    assert.equal(req.uri.query, null);
+  it('should keep query params null if only path params', (done) => {
+    const p = '/drive/v2/files/123abc';
+    nock(Utils.baseUrl).get(p).reply(200);
+    localDrive.files.get({fileId: '123abc'}, (err, res) => {
+      if (err) {
+        return done(err);
+      }
+      assert.equal(Utils.getQs(res), null);
+      nock(Utils.baseUrl).get(p).reply(200);
+      remoteDrive.files.get({fileId: '123abc'}, (err2, res2) => {
+        if (err2) {
+          return done(err2);
+        }
+        assert.equal(Utils.getQs(res2), null);
+        done();
+      });
+    });
   });
 
-  it('should keep query params as is', () => {
-    let req =
-        localDrive.files.get({fileId: '123abc', hello: 'world'}, Utils.noop);
-    assert.equal(req.uri.query, 'hello=world');
-    req = remoteDrive.files.get({fileId: '123abc', hello: 'world'}, Utils.noop);
-    assert.equal(req.uri.query, 'hello=world');
+  it('should keep query params as is', (done) => {
+    const p = '/drive/v2/files/123abc?hello=world';
+    nock(Utils.baseUrl).get(p).reply(200);
+    localDrive.files.get({fileId: '123abc', hello: 'world'}, (err, res) => {
```

Comment:
any reason not to use `pify` here like you do elsewhere?

nit: it would have been better if the pify changes were not included in this PR. It bundles optional changes with necessary changes making the patch more work to review.

Reply:
I only updated the tests where I would have had to add substantial code otherwise.  In many places, we didn't need to follow through on the full request, because we were just checking properties on the returned req object.  Now that it doesn't return a request (rather a promise), it was A LOT of adding nock and pify to keep callback depth sane.

Reply:
ack.